Cannot create aliased fixed array option

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

**V doctor:**

```
V full version: V 0.4.8 5ec9bb5.9e71e32
OS: linux, Linux version 6.6.8-arch1-1 (linux@archlinux) (gcc (GCC) 13.2.1 20230801, GNU ld (GNU Binutils) 2.41.0) #1 SMP PREEMPT_DYNAMIC Thu, 21 Dec 2023 19:01:01 +0000
Processor: 16 cpus, 64bit, little endian, 12th Gen Intel(R) Core(TM) i5-1240P

getwd: /home/ge
vexe: /home/ge/.vlang/v
vexe mtime: 2024-12-14 17:38:37

vroot: OK, value: /home/ge/.vlang
VMODULES: OK, value: /home/ge/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.47.1
Git vroot status: weekly.2024.50-36-g9e71e324
.git/config present: true

CC version: cc (GCC) 14.2.1 20240910
emcc version: N/A
thirdparty/tcc status: thirdparty-linux-amd64 0134e9b9
```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg create_aliased_fixed_array_option.vv`

```v
type Byt = [3]u8

fn main() {
	b := ?Byt(none)
	println(b)
}
```

**What did you expect to see?**

`Option(none)` in output.

**What did you see instead?**

```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/create_aliased_fixed_array_option.01JF35B5QP42PDT9WMGYCBF5KX.tmp.c:5843: error: cannot convert 'extern struct IError' to 'struct _option_main__Byt'
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

<sub><a href="https://huly.app/guest/vlang-66f40c4d-a476b54c67-771fdd?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzVkYzhhNTFlMDYzMmRkMDhlMzllNmMiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InctYWxleGFuZGVyLXZsYW5nLTY2ZjQwYzRkLWE0NzZiNTRjNjctNzcxZmRjIn0.ijTA-SXSMWgsfoURVW6jgD_KiDRGnRhXjizxPGvDlHs">Huly&reg;: <b>V_0.6-21601</b></a></sub>

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -5364,10 +5364,14 @@ fn (mut g Gen) cast_expr(node ast.CastExpr) {
 		g.expr(node.expr)
 		g.write('))')
 	} else if sym.kind == .alias && g.table.final_sym(node.typ).kind == .array_fixed {
-		if node.expr is ast.ArrayInit && g.assign_op != .decl_assign {
-			g.write('(${g.styp(node.expr.typ)})')
+		if node.typ.has_flag(.option) {
+			g.expr_with_opt(node.expr, expr_type, node.typ)
+		} else {
+			if node.expr is ast.ArrayInit && g.assign_op != .decl_assign {
+				g.write('(${g.styp(node.expr.typ)})')
+			}
+			g.expr(node.expr)
 		}
-		g.expr(node.expr)
 	} else if expr_type == ast.bool_type && node.typ.is_int() {
 		styp := g.styp(node_typ)
 		g.write('(${styp}[]){(')

--- a/vlib/v/tests/alias_cast_to_fixed_array_test.v
+++ b/vlib/v/tests/alias_cast_to_fixed_array_test.v
@@ -1,0 +1,24 @@
+type Bytes = [3]u8
+type Strs = [3]string
+
+fn test_none() {
+	b := ?Bytes(none)
+	println(b)
+	assert b == none
+
+	c := ?Strs(none)
+	println(c)
+	assert c == none
+}
+
+fn test_non_none() {
+	b := ?Bytes([u8(1), 2, 3]!)
+	println(b)
+	assert b != none
+	assert b?[2] == 3
+
+	c := ?Strs(['a', 'b', 'c']!)
+	println(c)
+	assert c != none
+	assert c?[2] == 'c'
+}

